packages/jest-cli/src/watch.js

```diff
@@ -60,10 +60,8 @@ const watch = (
   let testWatcher;
   let displayHelp = true;
 
-  hasteMap.on('change', ({eventsQueue, hasteFS, moduleMap}) => {
-    if (eventsQueue.find(({type}) => type !== 'change')) {
-      hasteContext = createHasteContext(config, {hasteFS, moduleMap});
-    }
```

Comment:
we always need a new instance of the hasteContext not just when they get added/deleted. At Facebook, if a module's "@providesModule" header changes, the module receives a new name for our "haste" module system. cc @rogeliog